tests/components/abode/test_binary_sensor.py

```diff
@@ -0,0 +1,39 @@
+"""Tests for the Abode binary sensor device."""
+from homeassistant.components.abode import ATTR_DEVICE_ID
+from homeassistant.components.abode.const import ATTRIBUTION
+from homeassistant.components.binary_sensor import (
+    DEVICE_CLASS_DOOR,
+    DOMAIN as BINARY_SENSOR_DOMAIN,
+)
+from homeassistant.const import (
+    ATTR_ATTRIBUTION,
+    ATTR_DEVICE_CLASS,
+    ATTR_FRIENDLY_NAME,
+    STATE_OFF,
+)
+
+from .common import setup_platform
+
+
+async def test_entity_registry(hass):
+    """Tests that the devices are registered in the entity registry."""
+    await setup_platform(hass, BINARY_SENSOR_DOMAIN)
+    entity_registry = await hass.helpers.entity_registry.async_get_registry()
+
+    entry = entity_registry.async_get("binary_sensor.front_door")
+    assert entry.unique_id == "2834013428b6035fba7d4054aa7b25a3"
+
+
+async def test_attributes(hass):
+    """Test the binary sensor attributes are correct."""
+    await setup_platform(hass, BINARY_SENSOR_DOMAIN)
+
+    state = hass.states.get("binary_sensor.front_door")
+    assert state.state == STATE_OFF
+    assert state.attributes.get(ATTR_ATTRIBUTION) == ATTRIBUTION
+    assert state.attributes.get(ATTR_DEVICE_ID) == "RF:01430030"
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=5184102)